rules/nat.go

```diff
@@ -18,64 +18,92 @@ import (
 	"fmt"
 	"sort"
 
+	"github.com/sirupsen/logrus"
+
 	"github.com/projectcalico/felix/iptables"
 )
 
+func makeRule(r *DefaultRuleRenderer, protocol string, action string, ipVersion uint8, rule []iptables.Rule) []iptables.Rule {
+	logrus.Debugf("\nprotocol  = %v action = %v\n", protocol, action)
```

Comment:
If these log lines are going to be left in (as opposed to being for your use during development, please use structured logging:
```
logrus.WithFields(logrus.Fields{"protocol": protocol, "action": action}).Debug("Making NAT outgoing rule")
```

- no new lines in the message (so that we get a standard format in the log file)
- use fields for including relevant data (makes it easier to parse the log by machine)
- clear message that says what the code is about to do (so if the next line panics we have an idea why).

Reply:
These logs are for my debugging. I will delete them before merging.